docs/content/release-notes/flink-1.15.md

```diff
@@ -0,0 +1,379 @@
+for keys which don't exist in the database. This is enabled by default.
+
+
+### [Support partition keys through metadata (for FileSystem connector) (FLINK-24617)](https://issues.apache.org/jira/browse/FLINK-24617)
+
+Now the formats implementing `BulkWriterFormatFactory` don't need to implement
+partition keys reading anymore, as it's managed internally by `FileSystemTableSource`.
+
+## Runtime & Coordination
+
+### [Implement file-based JobResultStore (FLINK-25431)](https://issues.apache.org/jira/browse/FLINK-25431)
+
+Introduces a file-based version of the `JobResultStore` with new configuration
+parameters `job-result-store.storage-path` and `job-result-store.delete-on-commit`. 
+This feature persists the final state of a job including its state of cleanup.
+
+### [Integrate retry strategy for cleanup stage (FLINK-25433)](https://issues.apache.org/jira/browse/FLINK-25433)
+
+Adds retry logic to the cleanup steps of a finished job. This feature changes the
+way Flink jobs are cleaned up. Instead of trying once to clean up the job, this
+step will be repeated until it succeeds. Users are meant to fix the issue that
+prevents Flink from finalizing the job cleanup. The retry functionality can be
+configured and disabled. More details can be found 
+[in the documentation](https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/deployment/config/#retryable-cleanup).
+
+### [Introduce common interfaces for cleaning up local and global job data (FLINK-25432)](https://issues.apache.org/jira/browse/FLINK-25432)
+
+Utilizes the newly added component `JobResultStore` (see 
+[FLINK-25431](https://issues.apache.org/jira/browse/FLINK-25431)) to pick up
+already finished but not properly cleaned up jobs after a failover.
+
+### [Introduce explicit shutdown signaling between `TaskManager` and `JobManager` (FLINK-25277)](https://issues.apache.org/jira/browse/FLINK-25277)
+
+Reduce down-scaling delay in reactive mode through signaling between `TaskManager` 
+and `JobManager`.
+
+### [FLIP-187: Adaptive Batch Scheduler (FLINK-24892)](https://issues.apache.org/jira/browse/FLINK-24892)
+
+In 1.15, we introduced a new scheduler to Flink: Adaptive Batch Scheduler. The 
+new scheduler can automatically decide parallelisms of job vertices for batch
+jobs, according to the size of data volume each vertex needs to process.
+
+More information about this feature and how to enable it could be found in 
+[the documentation](https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/elastic_scaling/#adaptive-batch-scheduler.)
+
+### [Release `TaskManagerJobMetricGroup` with the last slot rather than task (FLINK-24864)](https://issues.apache.org/jira/browse/FLINK-24864)
+
+TM-per-job metrics are now released upon last slot release rather than task, i.e. 
+may stay on TM longer.
+
+### [Issue in exceptions API(/jobs/:jobid/exceptions) in flink 1.13.2 (FLINK-25096)](https://issues.apache.org/jira/browse/FLINK-25096)
```

Comment:
@XComp please change the jira title to be more descriptive

Reply:
Good point - I updated the Jira issue title. Sorry for not thinking about it while losing the issue.

Reply:
Done ✅